moped-editor/src/views/projects/projectView/ProjectComponents/ComponentListItem.js

```diff
@@ -78,25 +78,27 @@ export default function ComponentListItem({
             <ZoomInIcon />
           </IconButton>
         </ListItemSecondaryAction>
-      </ListItem>
+      </ListItemButton>
       <Collapse in={isExpanded}>
-        <List component="div" disablePadding dense>
-          {!isComponentMapped && (
-            <ListItem dense className={classes.listItem}>
-              <ListItemText
-                primary={
-                  <Alert severity="error">Component is not mapped</Alert>
-                }
-              />
-            </ListItem>
-          )}
-          {component.description && (
-            <ListItem className={classes.nested}>
-              <ListItemText secondary={component.description} />
-            </ListItem>
-          )}
-          {additionalCollapseListItems}
-        </List>
+        {isExpanded ? (
```

Comment:
This prevents us from rendering the expanded content of every single component in the list, and it was the most noticeable improvement.

Reply:
great catch!